src/main/java/mops/businesslogic/file/FileServiceImpl.java

```diff
@@ -317,13 +316,40 @@ public Directory renameFile(Account account, long fileId, String newName) throws
         String[] fileNameParts = fileInfo.getName().split("\\.");
         if (fileNameParts.length > 1) {
             String fileExtension = "." + fileNameParts[fileNameParts.length - 1];
-            fileInfo.setName(newName + fileExtension);
+            newName = sanitizeName(newName + fileExtension);
+            fileInfo.setName(newName);
         } else {
             // file had no extension
+            newName = sanitizeName(newName);
             fileInfo.setName(newName);
         }
 
         fileInfoService.saveFileInfo(fileInfo);
         return directory;
     }
+
+    /**
+     * Sanitizes a filename.
+     *
+     * @param filename the filename
+     * @return sanitized String
+     */
+    @SuppressWarnings({ "PMD.AvoidLiteralsInIfCondition", "PMD.LawOfDemeter" })
+    private String sanitizeName(String filename) {
```

Comment:
If you give the new name and the extension as two separate parameters you would not have to split again.